quipucords/scanner/network/connect.py

```diff
@@ -339,13 +339,18 @@ def _handle_ssh_passphrase(credential):
 
     :param credential: The credential used for connections
     """
-    if (
-        credential.get("ssh_keyfile") is not None
-        and credential.get("ssh_passphrase") is not None
+    ssh_keyfile = credential.get("ssh_keyfile")
+    ssh_keyvalue = credential.get("ssh_keyvalue")
+
+    if credential.get("ssh_passphrase") is not None and (
+        ssh_keyfile is not None or ssh_keyvalue is not None
     ):
-        keyfile = credential.get("ssh_keyfile")
         passphrase = decrypt_data_as_unicode(credential["ssh_passphrase"])
-        cmd_string = f"ssh-add {keyfile}"
+        if ssh_keyfile is not None:
+            cmd_string = f'ssh-add "{ssh_keyfile}"'
+        else:
+            ssh_keyvalue = decrypt_data_as_unicode(ssh_keyvalue)
+            cmd_string = f"/bin/bash -c 'echo \"{ssh_keyvalue}\n\" | ssh-add -'"
```

Comment:
Not that I recall seeing.  Just ran again with an ssh key with a passphrase (the above if case), and the output has those there, but no private keys shown:

```
<192.168.30.131> ESTABLISH SSH CONNECTION FOR USER: abellotti
<192.168.30.131> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="abellotti"' -o ConnectTimeout=120 -o 'ControlPath="/Users/abellotti/DevHome/init_scripts/.ansible/cp/803dbc9628"' -tt 192.168.30.131 'if [ -f /etc/machine-id ] ; then ls --full-time /etc/machine-id 2>/dev/null | grep -o '"'"'[0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\}'"'"' ; fi'
<192.168.30.131> (0, b'2023-01-19\r\n', b'Shared connection to 192.168.30.131 closed.\r\n')
```

Also looked at /var/log/system.log and ~/.ansible/, not there either.

Reply:
also, looking at `pexpect.spawn`, it does not look like it would log unless the logfile parameter is specified. so I think we're good.  We can always change this code in the future to generate a temporary ssh_key file for running the ssh-add.